src/swupd/files_test.go

```diff
@@ -0,0 +1,198 @@
+package swupd
+
+import "testing"
+
+func TestTypeFromFlagFile(t *testing.T) {
+	testCases := []struct {
+		flag     byte
+		expected ftype
+	}{
+		{'F', FILE},
+		{'D', DIRECTORY},
+		{'L', LINK},
+		{'M', MANIFEST},
+		{'.', TYPE_UNSET},
+	}
+
+	for _, tc := range testCases {
+		t.Run(string(tc.flag), func(t *testing.T) {
+			f := File{}
+			var err error
+			if f.Type, err = typeFromFlag(tc.flag); err != nil {
+				t.Errorf("failed to set %v type flag on file", tc.flag)
+			}
+
+			if f.Type != tc.expected {
+				t.Errorf("file type was set to %v from %v flag", f.Type, tc.flag)
+			}
+		})
+	}
+
+	// space is never valid
+	t.Run("' '", func(t *testing.T) {
```

Comment:
I'm confused how this isn't failing as the value being passed is a string and not a byte (should just be `' '` I think).